### PR TITLE
More Quick Construction Recipes And Upgrades

### PR DESCRIPTION
**Pull request by gluesniffler**
# Description

This PR adds a few new recipes to Quick Construction (accessed by pressing Z on commonly used construction materials/items, such as rods, glass, steel, plasma, etc etc). Along with simple nesting for grouping items up under categories, akin to how the RCD works. Which allows engineers to massively speed up pipe laying work, disposal rebuilding, setting up the supermatter, etc etc.

First time doing UI code, apologies for the shitcode in advance :godo:

---

<details><summary><h1>Media</h1></summary>
<p>

https://github.com/user-attachments/assets/9d540d7f-7e4f-4a43-874d-5ea069011a37

</p>
</details>

---

# Considerations

Is this powergamey? Probably. Is it fucking awesome? Hell yeah. Some servers might question the fact that everyone can build stuff this fast now that they dont have to deal too much with the construction menu for the most common things... And might want to lock it behind traits or given as a job freebie, like CPR for doctors. 

Other servers might appreciate it if they are looking for more action to happen over the round, and giving everyone the ability to deal with breaches and repairs quicker is probably a step into that direction. I'll be observing for feedback and adjust accordingly.

It also shouldn't invalidate any eventual RPDs, or other devices of the kind since this is just a shortcut for construction, you still require materials and a doafter for most of them. 

---

# Changelog

:cl: Mocho
- add: Added a lot of recipes to the quick construction menus. Give it a shot by pressing Z with different construction materials in your hand!

### CODE REVIEW

**Comment by gluesniffler** [comment]
YAML Linter my beloathed... fixing.

File: Content.Shared/ShortConstruction/ShortConstructionComponent.cs [hunk 1]
```diff
@@ -2,14 +2,38 @@
...
+}
+
+[DataDefinition]
+public sealed partial class ShortConstructionEntry
+{
+    [DataField("prototype")]
+    public ProtoId<ConstructionPrototype>? Prototype { get; set; }
+
+    [DataField("category")]
+    public ShortConstructionCategory? Category { get; set; }
```
Comment (VMSolidus) [inline Content.Shared/ShortConstruction/ShortConstructionComponent.cs:23]:
```suggestion
    [DataField]
    public ProtoId<ConstructionPrototype>? Prototype { get; set; }

    [DataField]
    public ShortConstructionCategory? Category { get; set; }
```

File: Content.Shared/ShortConstruction/ShortConstructionComponent.cs [hunk 2]
```diff
@@ -2,14 +2,38 @@
...
+public sealed partial class ShortConstructionCategory
+{
+    [DataField("name")]
+    public string Name { get; set; } = string.Empty;
+
+    [DataField("icon")]
+    public SpriteSpecifier Icon { get; set; } = default!;
+
+    [DataField("entries")]
+    public List<ShortConstructionEntry> Entries { get; set; } = new();
```
Comment (VMSolidus) [inline Content.Shared/ShortConstruction/ShortConstructionComponent.cs:36]:
```suggestion
    [DataField]
    public string Name { get; set; } = string.Empty;

    [DataField]
    public SpriteSpecifier Icon { get; set; } = default!;

    [DataField]
    public List<ShortConstructionEntry> Entries { get; set; } = new();
```

File: Content.Client/ShortConstruction/UI/ShortConstructionMenuBUI.cs [hunk 3]
```diff
@@ -39,8 +39,19 @@ public ShortConstructionMenuBUI(EntityUid owner, Enum uiKey) : base(owner, uiKey
...
+            HorizontalExpand = true,
+            VerticalExpand = true,
+            BackButtonStyleClass = "RadialMenuBackButton",
+            CloseButtonStyleClass = "RadialMenuCloseButton"
+        };
+
+        if (_entManager.TryGetComponent<ShortConstructionComponent>(Owner, out var crafting))
+        {
+            CreateMenu(crafting.Entries);
+        }
```
Comment (VMSolidus) [inline Content.Client/ShortConstruction/UI/ShortConstructionMenuBUI.cs:53]:
```suggestion
        if (_entManager.TryGetComponent<ShortConstructionComponent>(Owner, out var crafting))
            CreateMenu(crafting.Entries);
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]

**Review by VMSolidus — APPROVED** [review]
https://github.com/user-attachments/assets/f168863f-ac69-4605-9071-7f5fed63df33

I've tested this ingame to verify that this feature works correctly and as advertised. I will say that this makes me a little giddy as someone who was once an engineer main. It means that proper "Combat engineering" is entirely viable since you can skip the clunky construction menu and go straight to building barriers.

**Review by FoxxoTrystan — APPROVED** [review]

**Comment by FoxxoTrystan** [comment]
oops removed solidus review

**Pull request merged**